pdns/zonemd.cc

```diff
@@ -0,0 +1,187 @@
+#include "zonemd.hh"
+
+#include "dnsrecords.hh"
+#include "dnssecinfra.hh"
+#include "sha.hh"
+#include "zoneparser-tng.hh"
+
+typedef std::pair<DNSName, QType> RRSetKey_t;
+typedef std::vector<std::shared_ptr<DNSRecordContent>> RRVector_t;
+
+struct CanonRRSetKeyCompare : public std::binary_function<RRSetKey_t, RRSetKey_t, bool>
+{
+  bool operator()(const RRSetKey_t& a, const RRSetKey_t& b) const
+  {
+    // FIXME surely we can be smarter here
```

Comment:
I believe this is my FIXME, I'll have a go at it

Reply:
Ah yes, can't use `std::tie` because of the custom comparison function. I'll leave it for now..